### PR TITLE
Expose restore tabs on startup option

### PR DESCRIPTION
**Pull request by jeremypw**
At the moment the only way to turn off restoring tabs in Files without turning of History for *all* apps in the System Settings is to edit the gsettings using either `dconf-editor` or the CLI, which is non-obvious.

This PR adds another switch to the AppMenu to toggle this setting.  This is also useful for allowing non-technical users to workaround/diagnose some issues (see https://github.com/elementary/files/issues/2342#issuecomment-1761801959)

### CODE REVIEW

File: src/View/Widgets/AppMenu.vala [hunk 1]
```diff
@@ -83,10 +83,14 @@ public class Files.AppMenu : Gtk.Popover {
             action_name = "win.singleclick-select"
         };
 
-        var foldes_before_files = new Granite.SwitchModelButton (_("Sort Folders before Files")) {
+        var folders_before_files = new Granite.SwitchModelButton (_("Sort Folders before Files")) {
             action_name = "win.folders-before-files"
         };
 
+        var restore_tabs = new Granite.SwitchModelButton (_("Restore Tabs on Starting")) {
```
Comment (zeebok) [inline src/View/Widgets/AppMenu.vala:90]: Not sure if this wording is best? When I see "start-up" I think more the machine/OS starting up. I am almost tempted to keep it layman and suggest "Restore Tabs from Last Time" but I am not totally convinced that is a good alternative lol

Comment (jeremypw) [reply src/View/Widgets/AppMenu.vala:90]: How about "Restore Tabs when Launching"

Comment (danirabbit) [reply src/View/Widgets/AppMenu.vala:90]: I feel like "when Launching" might be too ambiguous. 

Web calls this "Restore Tabs on Startup" which would we could go with for consistency
Chrome calls it "Continue where you left off" which feels super ambiguous :grimacing:
Firefox calls this "Show Tabs from Last Time" which I think validates @zeebok here and might be the most clear

Comment (jeremypw) [reply src/View/Widgets/AppMenu.vala:90]: OK, I'll go with @zeebok 's suggestion then.

**Review by zeebok — APPROVED** [review]

**Comment by zeebok** [comment]
Seems like checks failing due to this compile error:

`../src/Application.vala:188.17-188.36: error: `GLib.ApplicationCommandLine.printerr_literal' is not available in gio-2.0 2.72.4. Use gio-2.0 >= 2.80`

**Comment by jeremypw** [comment]
@zeebok  Yes, this change originated a while back in #2307  but at the time checks passed - I am not sure what has changed since then to cause this failure.  @Marukesu Any ideas?

**Comment by Marukesu** [comment]
It seems to be a mismatch between the vapi and library version.

glib [added a wrapper function to the printerr_literal vfunc in 2.80](https://gitlab.gnome.org/GNOME/glib/-/commit/b9c90c3a3e17f067870703c51d818020436acb1b). but the vfunc is older.

**Comment by jeremypw** [comment]
@Marukesu Thanks for the info.  Can it be changed back to `printerr` for now so it passes CI or can it be fixed by changing the workflow (in which case how)?  The tests pass when run on OS8.

**Comment by Marukesu** [comment]
Yes, there is no harm in changing it to cmd.printerr().

**Pull request merged**